src/navigation/context.py

```diff
@@ -167,11 +209,14 @@ def __init__(self):
         self.tf_listener = tf2_ros.TransformListener(self.tf_buffer)
         self.vel_cmd_publisher = rospy.Publisher("cmd_vel", Twist, queue_size=1)
         self.vis_publisher = rospy.Publisher("nav_vis", Marker, queue_size=1)
-        self.course_service = rospy.Service("course_service", mrover.srv.PublishCourse, self.recv_course)
+        self.enable_auton_service = rospy.Service("enable_auton", mrover.srv.PublishEnableAuton, self.recv_enable_auton)
         self.course = None
         self.rover = Rover(self)
         self.env = Environment(self)
 
-    def recv_course(self, req: mrover.srv.PublishCourseRequest) -> mrover.srv.PublishCourseResponse:
-        self.course = Course(self, req.course)
-        return mrover.srv.PublishCourseResponse(True)
+    def recv_enable_auton(self, req: mrover.srv.PublishEnableAutonRequest) -> mrover.srv.PublishEnableAutonResponse:
+        enable_msg = req.enableMsg
+        if enable_msg.enable:
+            self.course = convert_and_get_course(self, enable_msg)
+        # TODO: disable auton on signal here.
```

Comment:
Is this worth doing in this PR? Wouldn't it just be `else: self.course = None`?

Reply:
> Is this worth doing in this PR? Wouldn't it just be `else: self.course = None`?

Nah it would need to change the state to off (since that would be the signal to disable auton) which doesn't exist yet I have another member working on it right now though.

Reply:
gotcha